### DISCUSSION TITLE
width should include  two  storkeWidth

### DISCUSSION THREAD

**Discussion by vaynevayne** [fabricjs/fabric.js#10101] [General]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.2

### Node Version (if applicable)

None

### Link To Reproduction

https://codepen.io/V-alar/pen/mdZKQNq?editors=1010

### Steps To Reproduce

width = 100 , strokeWidth: 50 
like `box-sizing : content-box` the  computed width should is 50 + 100 + 50
now it only : 100 + 50 
![image](https://github.com/user-attachments/assets/05920765-53b4-489b-b12d-2aa3577e9245)

this is a bug? 
how  to fix it ? 

### Expected Behavior

like `box-sizing : content-box` the  computed width should is 50 + 100 + 50
like `box-sizing :  border-box` the  computed width should is 100

![image](https://github.com/user-attachments/assets/8a06c765-ec77-420d-a507-0795141e50ee)

### Actual Behavior

now the width is 150

**Comment by zhe-he** [comment]
<img width="321" alt="image" src="https://github.com/user-attachments/assets/305139a6-8738-4b54-a2b1-536260daf906">

因为是中部描边，所以是 100 + 50 / 2 + 50 / 2

**Comment by vaynevayne** [comment]
有办法修改什么东西,来达到我要的那个值吗? @zhe-he

**Comment by zhe-he** [comment]
> 有办法修改什么东西,来达到我要的那个值吗? @zhe-he

你用完整的demo可以实现你的需求，左上角可以把demo下载到本地。

https://github.com/fabricjs/fabric.js/discussions/9727

**Comment by vaynevayne** [comment]
不修改原型能实现吗? 可以的话 我想试一试你这个

另外 你不提一个pr 合到fabric吗?

**Reply by zhe-he** [reply]
不修改原型可以实现，简单的demo就是（如果不关心尺寸的话）。
如果关心尺寸，需要修改原型。

关于pr的问题，相关讨论在这里
https://github.com/fabricjs/fabric.js/discussions/9805
